### DISCUSSION TITLE
High Resolution Tiles & SVG input > 65536

### DISCUSSION THREAD

**Discussion by DhammaCharts** [libvips/libvips#2826] [Help wanted]
Hi,

### 1) High Resolution Tiles 

I would like to make @2x.png or @3x.png tiles using `dzsave`. I've used the following command:

`vips dzsave 'input.svg[scale=8]'  tiledir --layout google --centre --suffix .png --vips-progress`

To create [this map of a chart](https://www.dhammacharts.org/suttapitaka-chart/), the repo for this app is  [here](https://www.dhammacharts.org/suttapitaka-chart/).

I've been facing difficulty with slight blurriness on high resolution screens like retina. I've seen solutions online mentioning the @2x.png or @3x.png format and that I could create a different sets of HD tiles to be serves on such display. 

I could not find any mention of PNG resolution in VIPS documentation. Is there a way to proceed to have HD tiles ? I thought about making larger tiles `--tile-size 1024` and then resizing them smaller with high resolution, but I was not sucessful.

### 2)  SVG input > 65536

My other concern is about creating deeper pyramids for bigger charts. When scaling chart with the same command, I quickly hit errors because SVG load and cairo cannot handle bigger sizes. Is there a way to go beyond this limitation, knowing that my source image is SVG ?

Thank you very much

**Comment by jcupitt** [comment]
Hello @DhammaCharts,

1) You can make tiles any size you like. I think the "@2x.png" stuff comes from the way you organise the tiles on your webserver, and that's not something dzsave can control. 

2) Sadly, librsvg now has a very strict 64k x 64k limit. We'd need to integrate some other SVG library if we wanted to render larger images.

**Reply by DhammaCharts** [reply]
Hi, and thank you for your quick reply.

1. OK, I'll investigate further into that, it seems that @2x.png is simply a version with twice as much pixels on the sides of the image, so twice bigger. 
2. One way I found to work around it is to manually tile my SVG by changing the viewbox parameters, let say in 16 tiles, to create 16 SVGs and then apply dzsave to the first one, second one, etc. The problem of course is to join the resulting tiles together... which are in different folders, but it might be possible to load them at different coordinates on the final "map".

I've seen also during my research that TIFF images are more suitable for this kind of thing and I was wondering if SVG could be tiled first, then every tile exported to 64k x 64k TIFF (or VIPS images), then joined together in a big image, then finally tiled by dzsave. 

Just some thought ;-) thank you very much for you time and attention,
May you be well.

**Reply by jcupitt** [reply]
On 2., libvips used to fake >64k pixel images by moving a viewbox around (in the way you say), but that no longer works with librsvg either :( You'll find all the clipping breaks.

It's no longer possible to generate large images from SVG with librsvg, as far as I know.
